File: C/02_casino.c
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <stdlib.h>
#ifdef _WIN32
#include <Windows.h>
#else
#include <unistd.h>
#endif

void msg(char m[20])
{
    int i;
    printf("\n");
    for (i = 0; i < strlen(m) + 4; i++)
        printf("_");
    printf("\n\n  %s  \n", m);
    for (i = 0; i < strlen(m) + 4; i++)
        printf("_");
}

void end(int init, int now)
{
    if (init == now)
        printf("\n\nYou have no gain and no loss.");
    else if (init > now)
        printf("\n\nYou lost %d", init - now);
    else
        printf("\n\nYou gained %d", now - init);
}

int main()
{
    int init_plr, amt_plr, bet_plr, num_plr;
    int amt_dlr, bet_dlr, num_dlr, num_luck;
    printf("\nYour balance will be the double of what you bet. ");
    printf("\n\nEnter the bet amount : $ ");
    scanf("%d", &bet_plr);
    bet_dlr = 100 * ((rand() % 6) + 5);
    if (bet_plr > 1000)
    {
        printf("\nMax amount is 1000");
        bet_plr = 1000;
    }
    else if (bet_plr < 500)
    {
        printf("\nMin amount is 500");
        bet_plr = 500;
    }

    init_plr = amt_plr = 2 * bet_plr;
    amt_dlr = 2 * bet_dlr;
    printf("\n\nPlayer Bet Amount : $ %d", bet_plr);
    printf("\nDealer Bet Amount : $ %d", bet_dlr);

    while (1)
    {
        printf("\n\nEnter your number (1-8) : ");
        scanf("%d", &num_plr);
        if (num_plr > 8)
        {
            printf("\nTop number is 8");
            num_plr = 8;
        }
        else if (num_plr < 1)
        {
            printf("\nLowest number is 1");
            num_plr = 1;
        }

        printf("\n\nPlayer's Number : %d", num_plr);

        while (1)
        {
            num_dlr = (rand() % 8) + 1;
            if (num_dlr != num_plr)
                break;
        }

        printf("\nDealer Number : %d", num_dlr);

        printf("\n\nRolling . . .");
        printf(".");
        num_luck = (rand() % 8) + 1;
        sleep(2);
        printf("\n\nDrawn Number : %d", num_luck);
        if (num_plr == num_luck)
        {
            amt_plr += bet_dlr;
            amt_dlr -= bet_dlr;
            msg("PLAYER WON !");
            if (amt_dlr <= 0)
            {
                printf("\nDealer lost all his money");
                end(init_plr, amt_plr);
                break;
            }
        }
        else if (num_dlr == num_luck)
        {
            amt_plr -= bet_plr;
            amt_dlr += bet_dlr;
            msg("DEALER WON !");
            if (amt_plr <= 0)
                printf("\nYour balance is 0.");
                end(init_plr, 0);
                break;
        }
        else
            msg("DRAW !");

        printf("\n\n\nPlayer's Balance : $ %d", amt_plr);
        printf("\nDealer's Balance : $ %d", amt_dlr);
        printf("\n\n1. Continue \t 2. Quit \t : ");
        int cont;
        scanf("%d", &cont);
        printf("___________________________________________________");
        if (cont == 2)
        {
            end(init_plr, amt_plr);
            break;
        }
    }
    return 0;
}

File: Dart/02_casino.dart
import 'dart:io';
import 'dart:math';

Random random = new Random();

void msg(m) {
  int i;
  stdout.write("\n");
  for (i = 0; i < m.length + 4; i++) stdout.write("_");
  stdout.write("\n\n  ${m}  \n");
  for (i = 0; i < m.length + 4; i++) stdout.write("_");
}

void end(init, now) {
  if (init == now)
    stdout.write("\n\nYou have no gain and no loss.");
  else if (init > now)
    stdout.write("\n\nYou lost ${init - now}");
  else
    stdout.write("\n\nYou gained ${now - init}");
}

void main() {
  int init_plr, amt_plr, bet_plr, num_plr;
  int amt_dlr, bet_dlr, num_dlr, num_luck;
  stdout.write("\nYour balance will be the double of what you bet. ");
  stdout.write("\n\nEnter the bet amount : \$ ");
  bet_plr = int.parse(stdin.readLineSync());
  bet_dlr = 100 * (random.nextInt(5) + 5);
  if (bet_plr > 1000) {
    stdout.write("\nMax amount is 1000");
    bet_plr = 1000;
  } else if (bet_plr < 500) {
    stdout.write("\nMin amount is 500");
    bet_plr = 500;
  }

  init_plr = amt_plr = 2 * bet_plr;
  amt_dlr = 2 * bet_dlr;
  stdout.write("\n\nPlayer Bet Amount : \$ ${bet_plr}");
  stdout.write("\nDealer Bet Amount : \$ ${bet_dlr}");
  while (true) {
    stdout.write("\n\nEnter your number (1-8) : ");
    num_plr = int.parse(stdin.readLineSync());
    if (num_plr > 8) {
      stdout.write("\nTop number is 8");
      num_plr = 8;
    } else if (num_plr < 1) {
      stdout.write("\nLowest number is 1");
      num_plr = 1;
    }

    stdout.write("\n\nPlayer's Number : ${num_plr}");

    while (true) {
      num_dlr = random.nextInt(8) + 1;
      if (num_dlr != num_plr) break;
    }

    stdout.write("\nDealer Number : ${num_dlr}");

    stdout.write("\n\nRolling . . .");
    num_luck = random.nextInt(8) + 1;
    sleep(const Duration(seconds: 2));
    stdout.write("\n\nDrawn Number : ${num_luck}");
    if (num_plr == num_luck) {
      amt_plr += bet_dlr;
      amt_dlr -= bet_dlr;
      msg("PLAYER WON !");
      if (amt_dlr <= 0) {
        stdout.write("\nDealer lost all his money");
        end(init_plr, amt_plr);
        break;
      }
    } else if (num_dlr == num_luck) {
      amt_plr -= bet_plr;
      amt_dlr += bet_dlr;
      msg("DEALER WON !");
      if (amt_plr <= 0) stdout.write("\nYour balance is 0.");
      end(init_plr, 0);
      break;
    } else
      msg("DRAW !");

    stdout.write("\n\n\nPlayer's Balance : \$ ${amt_plr}");
    stdout.write("\nDealer's Balance : \$ ${amt_dlr}");
    stdout.write("\n\n1. Continue \t 2. Quit \t : ");
    int cont = int.parse(stdin.readLineSync());
    stdout.write("___________________________________________________");
    if (cont == 2) {
      end(init_plr, amt_plr);
      break;
    }
  }
}
